Playmate/View Controllers/Home Tab/MapPinsViewController.m

```diff
@@ -28,16 +35,25 @@ @implementation MapPinsViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
     
+    self.appliedFilters = NO;
+    self.filters = nil;
+    
     self.mapView.delegate = self;
     self.mapView.showsUserLocation = YES;
     
+    self.currentAnnotations = [[NSMutableArray alloc] init];
+    
     isFirstTimeGettingLocation = YES;
     
     [CLLocationManager locationServicesEnabled];
     
     [self initPinLocationManager];
     
-    [self fetchData];
+    if (self.appliedFilters == YES) {
+        [self fetchDataWithFilters:self.filters];
```

Comment:
I feel you do not need to pass the filters, because `self.filters` are global property.